autofit/graphical/factor_graphs/factor.py

```diff
@@ -198,13 +202,80 @@ def __init__(
             jacfwd=jacfwd,
         )
 
-    # @property
-    # def args(self):
-    #     return self._args
-
-    # @property
-    # def arg_names(self):
-    #     return self._arg_names
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        if self.plates:
+            return self._factor.shape
+
+        return ()
+
+    def __getitem__(
+        self, plates_index: Dict[Plate, Union[List[int], range, slice]]
+    ) -> "Factor":
+        return self.subset(plates_index)
+
+    def subset(
+        self,
+        plates_index: Dict[Plate, Union[List[int], range, slice]],
+        plate_sizes: Optional[Dict[Plate, int]] = None,
+    ) -> "Factor":
+        if not self.plates:
+            return self
+
+        plate_sizes = plate_sizes or dict(zip(self.plates, self.shape))
+        index = Variable.make_indexes(self, plates_index, plate_sizes)
+        subset_factor = self._factor[index]
+        kws = self._subset_jacobian(subset_factor, index)
+
+        subset = ", ".join(map("{0.name}={1}".format, self.plates, map(np.size, index)))
+        kws["name"] = f"{self.name}[{subset}]"
+        kws["eps"] = self.eps
+        kws["factor_out"] = self.factor_out
+        kws["plates"] = self.plates
+
+        return Factor(subset_factor, *self.args, **kws)
+
+    def _subset_jacobian(self, subset_factor, index):
+        jac_kws = {"vjp": self._vjp}
+        if self._vjp:
+            factor_vjp = None
+            if self._factor_vjp != self._jax_factor_vjp:
+                try:
+                    factor_vjp = self._factor_vjp[index]
```

Comment:
Create a local function which executes this try except pattern on a generic object to avoid repetition